src/client/opamCommands.ml

```diff
@@ -2404,65 +2408,61 @@ let switch cli =
       Arg.(some (list string)) None
   in
   let descr =
-    mk_opt ["description"] "STRING"
+    mk_opt ~cli cli_legacy ["description"] "STRING"
       "Attach the given description to a switch when creating it. Use the \
        $(i,set-description) subcommand to modify the description of an \
        existing switch."
       Arg.(some string) None
   in
   let full =
-    mk_flag ["full"]
+    mk_flag ~cli cli_legacy ["full"]
       "When exporting, include the metadata of all installed packages, \
        allowing to re-import even if they don't exist in the repositories (the \
        default is to include only the metadata of pinned packages)."
   in
   let freeze =
-    mk_flag ["freeze"]
+    mk_flag ~cli (introduced_in cli2_1) ["freeze"]
       "When exporting, locks all VCS urls to their current commit, failing if \
        it can not be retrieved. This ensures that an import will restore the \
        exact state. Implies $(b,--full)."
   in
   let no_install =
-    mk_flag ["no-install"]
+    mk_flag ~cli cli_legacy ["no-install"]
       "When creating a local switch, don't look for any local package \
        definitions to install."
   in
   let deps_only =
-    mk_flag ["deps-only"]
+    mk_flag ~cli cli_legacy ["deps-only"]
       "When creating a local switch in a project directory (i.e. a directory \
        containing opam package definitions), install the dependencies of the \
        project but not the project itself."
   in
   let force =
-    mk_flag ["force"]
+    mk_flag ~cli (introduced_in cli2_1) ["force"]
       "Only for $(i,set-invariant): force setting the invariant, bypassing \
        consistency checks."
   in
   let no_action =
-    mk_flag ["n"; "no-action"]
+    mk_flag ~cli (introduced_in cli2_1) ["n"; "no-action"]
       "Only for $(i,set-invariant): set the invariant, but don't enforce it \
        right away: wait for the next $(i,install), $(i,upgrade) or similar \
        command."
   in
   (* Deprecated options *)
-  let d_alias_of =
-    mk_opt ["A";"alias-of"]
-      "COMP"
-      "This option is deprecated."
-      Arg.(some string) None
+  let d_alias_of = Term.pure None
+(*
+    mk_opt ~cli (removed_since cli2_0 cli2_0 ~replaced:"opam switch <switch-name> <compiler>")
+      ["A";"alias-of"] "COMP" "" Arg.(some string) None
+*)
```

Comment:
Similarly to `--no-aspcud`, this can be restored just to make the 2.0 behaviour of a warning consistent!